tests/core/uc20-recovery/task.yaml

```diff
@@ -0,0 +1,127 @@
+summary: verify booting into recovery mode on UC20 systems
+
+systems: [ ubuntu-core-20-* ]
+
+restore: |
+  rm -f /writable/systems.json.run
+  rm -f /writable/systems.label
+
+  if mountpoint /usr/sbin/shutdown; then
+      umount /usr/sbin/shutdown
+  fi
+
+debug: |
+  if [ -e systems.json ]; then
+      cat systems.json
+  fi
+  if [ -e /tmp/mock-shutdown.calls ]; then
+      cat /tmp/mock-shutdown.calls
+  fi
+
+execute: |
+  if [ "$SPREAD_REBOOT" == "0" ]; then
+      echo "In run mode"
+
+      snap install jq
+      # devmode as the snap does not have snapd-control
+      snap install test-snapd-curl --devmode --edge
+
+      MATCH 'snapd_recovery_mode=run' < /proc/cmdline
+      # verify we are in run mode via the API
+      test-snapd-curl.curl -s --unix-socket /run/snapd.socket http://localhost/v2/system-info > system-info
+      jq -r '.result["system-mode"]' < system-info | MATCH 'run'
+
+      echo "Obtain available systems"
+      test-snapd-curl.curl -s --unix-socket /run/snapd.socket http://localhost/v2/systems > systems.json
+      # TODO:UC20: there is only one system for now
+      jq .result.systems[0].current < systems.json | MATCH 'true'
+      label="$(jq -r .result.systems[0].label < systems.json)"
+      test -n "$label"
+      # make sure that the seed exists
+      test -d "/var/lib/snapd/seed/systems/$label"
+      jq -r .result.systems[0].actions[].mode < systems.json | MATCH 'recover'
+
+      # keep a copy of the systems dump for later reference
+      cp systems.json /writable/systems.json.run
+      echo "$label" > /writable/systems.label
+
+      # redirect shutdown command to our mock to observe calls and avoid racing
+      # with spread
+      mount -o bind "$PWD/mock-shutdown" /usr/sbin/shutdown
```

Comment:
nice work, thanks for this, perhaps we should try to fix this elsewhere we expect snapd started shutdown's to be done as well?

Reply:
not for this PR obviously